components/core/preview.tsx

```diff
@@ -44,6 +56,16 @@ const Preview: React.FC<EditorProps> = ({ className }) => {
                 >
                     Sample data fetching
                 </Button>
+                <Button
+                    onClick={() => {
+                        handleSubmitIncrement();
+                        setIsSmallExploding(!isSmallExploding);
```

Comment:
On the first click, you set `isSmallExploding` to `true` and confetti pops. That's fine. On the second click, I expect confetti to pop but it doesn't because you're setting `isSmallExploding` to `false` in that second click, leading to no pop even though everything went as expected. A better pattern would be to introduce a new button click handler at the top of the file that handles submitIncrement and sets of explosion.